From 6b7c24cde514164c482d59e1eab50d227b616f44 Mon Sep 17 00:00:00 2001
From: Jammy Huang <jammy_huang@aspeedtech.com>
Date: Thu, 11 Sep 2025 11:42:52 +0800
Subject: [PATCH 21/32] board: ibex_ast2700: Enable pcie settings for all 27xx

All PCIE nodes should be available in all 27xx.

Signed-off-by: Jammy Huang <jammy_huang@aspeedtech.com>
Change-Id: Id8deee019d8e51e13206e1aa010e6492b31bd1e2
(cherry picked from commit f1a9f99d5d08a567c9819cae6e025e46cd72a2c6)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 board/aspeed/ibex_ast2700/ibex_ast2700.c | 41 ++++++++++++------------
 1 file changed, 21 insertions(+), 20 deletions(-)

diff --git a/board/aspeed/ibex_ast2700/ibex_ast2700.c b/board/aspeed/ibex_ast2700/ibex_ast2700.c
index d9ea9e3c26..a544e0abb4 100644
--- a/board/aspeed/ibex_ast2700/ibex_ast2700.c
+++ b/board/aspeed/ibex_ast2700/ibex_ast2700.c
@@ -166,22 +166,22 @@ static int pci_init(void)
 		return 0;
 	}
 
-	if (efuse == 0) {
-		// setup preset for plda2
-		writel(0x12600000, (void *)ASPEED_PLDA2_PRESET0);
-		writel(0x00012600, (void *)ASPEED_PLDA2_PRESET1);
-
-		// Enable Bridge2 MSI
-		clrbits_le32((void *)ASPEED_PLDA2_MSI_CAP, BIT(3));
-		// Set Bridge2 INTA
-		clrsetbits_le32((void *)ASPEED_PLDA2_MSI_CAP, GENMASK(2, 0), 0x1);
-
-		// clk/reset for e2m
-		setbits_le32(&scu->clkgate_clr, SCU_CPU_CLKGATE1_E2M1);
-		mdelay(10);
-		setbits_le32(&scu->modrst2_clr, SCU_CPU_RST2_E2M1);
-	}
+	/* cpu-die pcie node 1 */
+	// setup preset for plda2
+	writel(0x12600000, (void *)ASPEED_PLDA2_PRESET0);
+	writel(0x00012600, (void *)ASPEED_PLDA2_PRESET1);
+
+	// Enable Bridge2 MSI
+	clrbits_le32((void *)ASPEED_PLDA2_MSI_CAP, BIT(3));
+	// Set Bridge2 INTA
+	clrsetbits_le32((void *)ASPEED_PLDA2_MSI_CAP, GENMASK(2, 0), 0x1);
+
+	// clk/reset for e2m
+	setbits_le32(&scu->clkgate_clr, SCU_CPU_CLKGATE1_E2M1);
+	mdelay(10);
+	setbits_le32(&scu->modrst2_clr, SCU_CPU_RST2_E2M1);
 
+	/* cpu-die pcie node 0 */
 	// setup preset for plda1
 	writel(0x12600000, (void *)ASPEED_PLDA1_PRESET0);
 	writel(0x00012600, (void *)ASPEED_PLDA1_PRESET1);
@@ -191,16 +191,17 @@ static int pci_init(void)
 	// Set Bridge1 INTA
 	clrsetbits_le32((void *)ASPEED_PLDA1_MSI_CAP, GENMASK(2, 0), 0x1);
 
-	// Enable Bridge3 MSI
-	clrbits_le32((void *)ASPEED_PLDA3_MSI_CAP, BIT(3));
-	// Set Bridge3 INTA
-	clrsetbits_le32((void *)ASPEED_PLDA3_MSI_CAP, GENMASK(2, 0), 0x1);
-
 	// clk/reset for e2m
 	setbits_le32(&scu->clkgate_clr, SCU_CPU_CLKGATE1_E2M0);
 	mdelay(10);
 	setbits_le32(&scu->modrst2_clr, SCU_CPU_RST2_E2M0);
 
+	/* io-die pcie node */
+	// Enable Bridge3 MSI
+	clrbits_le32((void *)ASPEED_PLDA3_MSI_CAP, BIT(3));
+	// Set Bridge3 INTA
+	clrsetbits_le32((void *)ASPEED_PLDA3_MSI_CAP, GENMASK(2, 0), 0x1);
+
 	return 0;
 }
 
-- 
2.34.1

